src/Umbraco.Web.Common/Extensions/LinkGeneratorExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using Umbraco.Core;
+using Microsoft.AspNetCore.Routing;
+using System.Reflection;
+using Umbraco.Web.Common.Install;
+
+namespace Umbraco.Extensions
+{
+    public static class LinkGeneratorExtensions
+    {
+        /// <summary>
+        /// Return the back office url if the back office is installed
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns></returns>
+        public static string GetBackOfficeUrl(this LinkGenerator linkGenerator)
+        {
+
+            Type backOfficeControllerType;
+            try
+            {
+                backOfficeControllerType = Assembly.Load("Umbraco.Web.BackOffice")?.GetType("Umbraco.Web.BackOffice.Controllers.BackOfficeController");
+                if (backOfficeControllerType == null) return "/"; // this would indicate that the installer is installed without the back office
+            }
+            catch (Exception)
+            {
+                return "/"; // this would indicate that the installer is installed without the back office
```

Comment:
In theory this should be the virtual path, right?

Reply:
yep, will fix to use `return hostingEnvironment.ApplicationVirtualPath;`